app/controllers/npq_separation/admin/milestones_controller.rb

```diff
@@ -0,0 +1,97 @@
+class NpqSeparation::Admin::MilestonesController < NpqSeparation::AdminController
+  before_action :set_cohort
+  before_action :set_schedule
+  before_action :ensure_super_admin
+  before_action :set_milestone, only: %i[edit update destroy]
+  before_action :set_statement_date_options, only: %i[new create edit update]
+  before_action :set_declaration_types_available, only: %i[new create]
+
+  def new
+    @service = Milestones::Create.new(schedule_id: @schedule.id)
+  end
+
+  def create
+    @service = Milestones::Create.new(
+      schedule_id: @schedule.id,
+      declaration_type: params.fetch(:milestones_create, {})[:declaration_type],
+      statement_date: params.fetch(:milestones_create, {})[:statement_date],
+    )
```

Comment:
You could use the different handling of named params versus positional params to separate out the 'user params' from the 'url params'?

```ruby
Milestones::Create.new(@schedule.id, **params.fetch(:milestones_create, {}))
```

Any unexpected params would trigger an error?